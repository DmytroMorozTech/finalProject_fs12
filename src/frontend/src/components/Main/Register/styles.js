import { makeStyles } from '@material-ui/styles'

export default makeStyles(theme => ({
  register: {
    width: '100vw',
    height: '100vh',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.palette.background.default,
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 150,
    flexDirection: 'column'
  },
  registerPage: {
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  registerHeader: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 80,
    flexDirection: 'column',
    alignItems: 'center'
  },
  registerHeaderText: {
    fontSize: theme.typography.h1
  }
}))