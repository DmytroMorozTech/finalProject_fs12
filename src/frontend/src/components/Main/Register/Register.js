import React from 'react'
import styles from './styles'
import RegisterPage from './RegisterPage/RegisterPage'

const Register = () => {
  const classes = styles()
  return (
    <div className={classes.register}>
      <div className={classes.registerPage}>
        <RegisterPage/>
      </div>
    </div>
  )
}

export default Register