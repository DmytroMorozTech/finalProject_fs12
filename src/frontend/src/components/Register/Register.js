import React from 'react'
import styles from './styles'
import RegisterPage from './RegisterPage/RegisterPage'
import LinkedinLogo from '../../shared/LinkedinLogo/LinkedinLogo'
import RegisterGooglePage from './RegisterGooglePage/RegisterGooglePage'

const Register = (props) => {
  const {isByGoogle} = props
  const classes = styles()
  return (
    <div className={classes.register}>
      <div className={classes.registerHeader}>
        <LinkedinLogo/>
        <h1 className={classes.registerHeaderText}>Make the most of your professional life</h1>
      </div>
      <div className={classes.registerPage}>
        {isByGoogle ? <RegisterGooglePage/> : <RegisterPage/>}
      </div>
    </div>
  )
}

export default Register