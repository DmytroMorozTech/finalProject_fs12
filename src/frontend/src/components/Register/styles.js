import { makeStyles } from '@material-ui/styles'

export default makeStyles(theme => ({
  register: {
    width: '100vw',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.palette.background.default,
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 150,
    flexDirection: 'column'
  },
  registerPage: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  registerHeader: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: theme.spacing(5),
    flexDirection: 'column',
    alignItems: 'center'
  },
  registerHeaderText: {
    fontSize: theme.typography.h1.fontSize,
    fontWeight: theme.typography.h4.fontWeight
  }
}))