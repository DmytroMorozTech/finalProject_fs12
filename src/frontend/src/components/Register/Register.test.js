import { shallow } from 'enzyme'
import React from 'react'
import Register from './Register'

jest.mock('./styles', () => () => ({}))

const setUp = (props) => shallow(<Register {...props}/>)

describe('should render Register component with RegisterGooglePage', () => {
  let isByGoogle = true
  let component

  beforeEach(() => {
    component = setUp({ isByGoogle })
  })

  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should render RegisterGooglePage', () => {
    let wrapper = component.find('RegisterGooglePage')
    expect(wrapper.length).toBe(1)
  })
})

describe('should render Register component with RegisterPage', () => {
  let isByGoogle = false
  let component

  beforeEach(() => {
    component = setUp({ isByGoogle })
  })

  it('should render RegisterPage', () => {
    let wrapper = component.find('RegisterPage')
    expect(wrapper.length).toBe(1)
  })
})
