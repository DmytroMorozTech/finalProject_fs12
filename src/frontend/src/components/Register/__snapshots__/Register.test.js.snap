// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Register component with RegisterGooglePage to match snapshot 1`] = `
<div>
  <div>
    <LinkedinLogo />
  </div>
  <div>
    <RegisterGooglePage />
  </div>
</div>
`;
